src/plan_expand_hypertable.c

```diff
@@ -1173,14 +1190,22 @@ ts_plan_expand_hypertable_chunks(Hypertable *ht, PlannerInfo *root, RelOptInfo *
 		propagate_join_quals(root, rel, &ctx);
 
 	inh_oids = get_chunk_oids(&ctx, root, rel, ht);
+
+	/* nothing to do here if we have no chunks and no data nodes */
+	if (list_length(inh_oids) + list_length(ht->data_nodes) == 0)
+		return;
+
```

Comment:
This looks more like an optimization than a necessary change for PG13. IMHO, it is better to keep the changes down so that we do not accidentally introduce a bug and add optimizations like this to after PG13 compiles.

Reply:
This is required because expand_planner_arrays has an assertion for non-zero size change.